### PR TITLE
Feature/39148 pupil status UI pt2

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
- Added pupil status presenter, new markup and styles to support pupil status business expectations
- Added frontend functionality to expand/collapse details element when related card component is clicked

### CODE REVIEW

**Comment by GuyHarwood** [comment]
please disable incomplete checks lookup as discussed.

File: admin/spec/back-end/helpers/pupil-status-presenter.spec.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      checkEndDate: moment.utc().add(1, 'days')
+    }
+    it('collects pupils with errors', () => {
+      const pupils = [
+        {
+          status: 'Error in processing'
+        },
+        {
+          status: 'Incomplete'
+        }
```
Comment (jon-shipley) [inline admin/spec/back-end/helpers/pupil-status-presenter.spec.js]: There should be at least one more status, or it *might* not be filtering at all.  This comment seems to apply to other tests here too.

File: admin/views/pupil-status/pupil-status-details/error-details.ejs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            <dt>
+                <div><strong><u>Either</u> log into the device that pupil used</strong></div>
+                <div>
+                    <p>
+                        Step 1 <br/>
+                        Log into the device this pupil used to take the check and the Live MTC should send through automatically.
+                    </p>
+                    <p>
+                        Step 2 <br/>
+                        Check this <u>pupil status</u> page to see if this pupil is now listed as 'Complete'. If not follow the instructions below
```
Comment (jon-shipley) [inline admin/views/pupil-status/pupil-status-details/error-details.ejs]: Missing full-stop?

File: admin/views/pupil-status/pupil-status-details/error-details.ejs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          Steps to take
+        </span>
+    </summary>
+    <div class="govuk-details__text">
+        <dl class="govuk-panel-list">
+            <dt>
+                <div><strong><u>Either</u> log into the device that pupil used</strong></div>
+                <div>
+                    <p>
+                        Step 1 <br/>
```
Comment (jon-shipley) [inline admin/views/pupil-status/pupil-status-details/error-details.ejs]: Step 1 and Step 2 seem like they should be marked up with an ordered list

File: admin/views/pupil-status/pupil-status-details/error-details.ejs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+<details class="govuk-details govuk-!-margin-bottom-0" data-module="govuk-details">
+    <summary class="govuk-details__summary">
+        <span class="govuk-details__summary-text">
+          Steps to take
+        </span>
+    </summary>
+    <div class="govuk-details__text">
+        <dl class="govuk-panel-list">
+            <dt>
```
Comment (jon-shipley) [inline admin/views/pupil-status/pupil-status-details/error-details.ejs]: I see `dt` tags but not the corresponding `dd` tags.

File: admin/views/pupil-status/pupil-status-details/error-details.ejs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+                        Log into the device this pupil used to take the check and the Live MTC should send through automatically.
+                    </p>
+                    <p>
+                        Step 2 <br/>
+                        Check this <u>pupil status</u> page to see if this pupil is now listed as 'Complete'. If not follow the instructions below
+                    </p>
+                </div>
+            </dt>
+            <dt>
+                <div><strong><u>or</u> restart check for pupil</strong></div>
```
Comment (jon-shipley) [inline admin/views/pupil-status/pupil-status-details/error-details.ejs]: The `u` tag is now the Unarticulated Annotation Element and NOT underline.  I don't see why we would want to underline anyway - won't this get mistaken for a link?

File: admin/views/pupil-status/view-pupil-status.ejs [hunk 6]
```diff
@@ -2,37 +2,274 @@
...
+                </h1>
+                <div class="govuk-panel__body custom-card-small-text">
+                    of <%= pupilStatusViewData.totalPupilsCount %> pupils
+                </div>
+            </div>
+            <div id="orange-card" class="govuk-panel govuk-panel custom-card orange-card">
+                <div class="govuk-panel__body custom-card-small-text">
+                    Pupils that have not started yet
+                </div>
+                <h1 class="custom-card-large-text">
```
Comment (jon-shipley) [inline admin/views/pupil-status/view-pupil-status.ejs]: So the heading, for the whole page, is a Number?

File: admin/views/pupil-status/view-pupil-status.ejs [hunk 7]
```diff
@@ -2,37 +2,274 @@
...
+                </h1>
+                <div class="govuk-panel__body custom-card-small-text">
+                    of <%= pupilStatusViewData.totalPupilsCount %> pupils
+                </div>
+            </div>
+            <div id="blue-card" class="govuk-panel govuk-panel custom-card blue-card">
+                <div class="govuk-panel__body custom-card-small-text">
+                    Pupils not taking the check
+                </div>
+                <h1 class="custom-card-large-text">
```
Comment (jon-shipley) [inline admin/views/pupil-status/view-pupil-status.ejs]: Using multiple `H1` will make it very hard for screen reader users.  Ideally, there should be a single `H1` tag.

File: admin/views/pupil-status/view-pupil-status.ejs [hunk 8]
```diff
@@ -2,37 +2,274 @@
...
+                                    <li>A PIN has yet to be generated for pupil to start the check</li>
+                                    <li>A PIN has expired before use</li>
+                                </ul>
+                            </div>
+                        </dt>
+                        <dt>
+                            <div><strong>Not taking the check</strong></div>
+                            <div>
+                                A reason for "not taking the check" has been applied to the pupil.
+                                If removed the pupil returns to the previous state. To change this go to <a href="/pupils-not-taking-the-check/pupils-list">"Remove pupil from register"</a> page.
```
Comment (jon-shipley) [inline admin/views/pupil-status/view-pupil-status.ejs]: Please wrap long lines.

File: admin/spec/back-end/helpers/pupil-status-presenter.spec.js (hunk 1 above)
Comment (GeorgeChatzigiannis) [reply admin/spec/back-end/helpers/pupil-status-presenter.spec.js]: @jon-shipley added large data input for all included tests

File: admin/views/pupil-status/pupil-status-details/error-details.ejs (hunk 3 above)
Comment (GeorgeChatzigiannis) [reply admin/views/pupil-status/pupil-status-details/error-details.ejs]: I'l keep this as description list element as per the last commit until a change in the requirements.

Comment (GeorgeChatzigiannis) [reply admin/views/pupil-status/pupil-status-details/error-details.ejs]: resolved

File: admin/views/pupil-status/view-pupil-status.ejs [hunk 9]
```diff
@@ -2,37 +2,279 @@
...
+                                </ul>
+                            </div>
+                        </dt>
+                        <dt>
+                            <div><strong>Not taking the check</strong></div>
+                            <div>
+                                A reason for "not taking the check" has been applied to the pupil.
+                                If removed the pupil returns to the previous state.
+                                To change this go to <a href="/pupils-not-taking-the-check/pupils-list">"Remove pupil from register"</a> page.
+                                The pupils check is being sent to the MTC data base. For technical reasons this is taking longer than expected.
```
Comment (jon-shipley) [inline admin/views/pupil-status/view-pupil-status.ejs]: Is this wording about 'data base' correct?  If the pupil is not taking a check then there isn't a check to send to the database. 😕

**Pull request merged**